qemu/tests/virtio_serial_hotplug_port_pci.py

```diff
@@ -48,24 +81,44 @@ def run(test, params, env):
                 buses.append(device)
             else:
                 serial_devices.append(device)
-    vm.devices.simple_hotplug(buses[0], vm.monitor)
-    vm.devices.simple_hotplug(serial_devices[0], vm.monitor)
-    pre_nr = serial_devices[0].get_param('nr')
-    # Try hotplug different device with same 'nr'
-    serial_devices[1].set_param('bus', serial_devices[0].get_param('bus'))
-    serial_devices[1].set_param('nr', pre_nr)
-    try:
-        serial_devices[1].hotplug(vm.monitor)
-    except QMPCmdError as e:
-        if 'A port already exists at id %d' % pre_nr not in str(e.data):
-            test.fail('Hotplug fail for %s, not as expected' % str(e.data))
-    else:
-        test.fail('Hotplug with same "nr" option success while should fail')
-    serial_devices[1].set_param('nr', int(pre_nr) + 1)
-    vm.devices.simple_hotplug(serial_devices[1], vm.monitor)
-    for device in serial_devices:
-        add_virtio_ports_to_vm(vm, params, device)
-    params['file_transfer_serial_port'] = serials[0]
-    transfer_data(params, vm, sender='host')
-    params['file_transfer_serial_port'] = serials[1]
-    transfer_data(params, vm, sender='guest')
+
+    for i in range(repeat_times):
+        error_context.context("Hotplug/unplug serial devices the %s time"
+                              % (i+1), logging.info)
+        vm.devices.simple_hotplug(buses[0], vm.monitor)
+        vm.devices.simple_hotplug(serial_devices[0], vm.monitor)
+        pre_nr = serial_devices[0].get_param('nr')
+
+        # Try hotplug different device with same 'nr'
+        if params.get("plug_same_nr") == "yes":
+            serial_devices[1].set_param('bus', serial_devices[0].get_param('bus'))
+            serial_devices[1].set_param('nr', pre_nr)
+            try:
+                serial_devices[1].hotplug(vm.monitor)
+            except QMPCmdError as e:
+                if 'A port already exists at id %d' % pre_nr not in str(e.data):
+                    test.fail('Hotplug fail for %s, not as expected' % str(e.data))
+            else:
+                test.fail('Hotplug with same "nr" option success while should fail')
+        serial_devices[1].set_param('nr', int(pre_nr) + 1)
```

Comment:
This line should belong to the above `if` block, you don't need it for normal hotplug test.

Reply:
Updated, thanks